lib/expression/callbacks/standard.ex

```diff
@@ -87,19 +87,25 @@ defmodule Expression.Callbacks.Standard do
                     "day" => 31
                   },
                   result: ~D[2022-01-31]
+  @expression_doc doc: "Invalid date inputs",
+                  expression: "date(nil, nil, nil)",
+                  context: %{},
+                  result: %{
+                    "__type__" => "expression/v1error",
+                    "error" => true,
+                    "message" => "Invalid date"
+                  }
   def date(ctx, year, month, day) do
     [year, month, day] = eval_args!([year, month, day], ctx)
 
-    fields = [
-      calendar: Calendar.ISO,
-      year: year,
-      month: month,
-      day: day,
-      time_zone: "Etc/UTC",
-      zone_abbr: "UTC"
-    ]
-
-    struct(Date, fields)
+    with true <- is_integer(year),
+         true <- is_integer(month),
+         true <- is_integer(day),
+         {:ok, date} <- Date.new(year, month, day) do
+      date
+    else
+      _ -> Expression.error("Invalid date")
```

Comment:
Can we provide the supplied value as part of the error here? That'll put an author in a better position to diagnose the issue at hand.

Reply:
```suggestion
      _ -> Expression.error("Invalid date: date(#{inspect(year)}, #{inspect(month)}, #{inspect(day)})") 
```
or something similar?

Reply:
Yes, great suggestion @smn, done in https://github.com/turnhub/expression/pull/256/commits/12d8dcb0059c292c89013df6f1a1f80615105b69

Reply:
cool, we'll just need to update the test to match I think

Reply:
I added here: https://github.com/turnhub/expression/pull/256/commits/9f7fe414843beac7d567f3f9dd3dfdba3151d3a7